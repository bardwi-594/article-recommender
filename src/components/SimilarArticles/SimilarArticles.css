
div.list-item {
    margin: 30px;
}

div.grid-container {
    display: flex;
    flex-direction: row;
}

div.card {
    margin: 1rem auto;
    box-shadow: 6px 8px 22px #e1e1e3, -15px -15px 27px #ffffff;
    margin-bottom: 1.6%;
    background: #ffffff;
    font-family: 'Roboto', sans-serif;
    border-radius: 20px;
    overflow: hidden;
    z-index: 0;
    min-height: 150px;
    cursor: pointer;
}

div.grid-item-img img {
    display: block;
    position: relative;
    background-color: transparent;
    width: 100%;
    height: 100%;
}
div.grid-container>.grid-item-text {
    position: relative;
    display: inline-block;
}


div.grid-item-text {
    padding: 25px;
    flex: 1 1 auto;
    height: auto;
    position: relative;
}

div.grid-item-text h3.article-header {
    display: block;
    margin: 0;
    font-family: 'Roboto', sans-serif;
    font-weight: 500;
    color: rgba(0,0,0,.87);
    line-height: 1.285em;
    font-size: 1.6rem;
    width: 90%;
}

div.source-font, div.date-font {
    color: #959595;
    font-weight: 500;
}
div.similarity {
    color: #41bb56;
    text-transform: lowercase;
    font-weight: 700;
}

div.article-items {
    display: inline-block;
    padding: 2em 0 .25em 0;
    font-size: 1.3rem;
    font-family: 'Roboto', sans-serif;
    line-height: 1;
}

div.article-items>.article-item {
    display: inline-block;
    table-layout: fixed;
    list-style-type: none;
    list-style-position: outside;
    margin-right: 10px;
    margin-top: 4px;
    padding-right: 10px;
    border-right: 2px solid #959595;
}

div.article-items>.article-item:first-child {
    margin-left: 0;
    padding-left: 0;
}
div.article-items>.article-item:last-child {
    border-right:none;
}



@media (min-width: 500px){
    .cointainer {
        padding-left: 24px;
        padding-right: 24px;
    }
     
    div.card {
        flex-direction: row;
        max-width: 750px;
    }

    div.grid-item-img{
        flex-basis: 20%;
        height: auto;
    }

    div.grid-item-text {
        flex-basis: 80%;
    }
}

@media only screen and (min-width: 500px){
    div.card {
        flex-direction: row;
        max-width: 700px;
        min-height: 100px;
    }

    div.grid-item-img{
        flex-basis: 25%;
    }

    div.article-items {
        display: inline-block;
        padding: 2em 0 .25em 0;
        font-size: 0.7rem;
        font-family: 'Roboto', sans-serif;
        line-height: 1;
    }

    div.grid-item-text {
        flex-basis: 75%;
    }

    div.grid-item-text h3.article-header {
        display: block;
        margin: 0;
        font-family: 'Roboto', sans-serif;
        font-weight: 500;
        color: rgba(0,0,0,.87);
        line-height: 1.285em;
        font-size: 1rem;
        width: 90%;
    }

    div.article-items>.article-item {
        display: inline-block;
        table-layout: fixed;
        list-style-type: none;
        list-style-position: outside;
        margin-right: 6px;
        margin-top: 4px;
        padding-right: 6px;
        border-right: 2px solid #959595;
    }
}

@media only screen and (min-width: 280px){
    div.card {
        flex-direction: row;
        max-width: 700px;
        min-height: 120px;
    }

    div.grid-item-img{
        flex-basis: 25%;
    }

    div.article-items {
        display: inline-block;
        padding: 2em 0 .25em 0;
        font-size: 0.6rem;
        font-family: 'Roboto', sans-serif;
        line-height: 1;
    }

    div.grid-item-text {
        flex-basis: 75%;
    }

    div.grid-item-text h3.article-header {
        display: block;
        margin: 0;
        font-family: 'Roboto', sans-serif;
        font-weight: 500;
        color: rgba(0,0,0,.87);
        line-height: 1.285em;
        font-size: 0.8rem;
        width: 90%;
    }

    div.article-items>.article-item {
        display: inline-block;
        table-layout: fixed;
        list-style-type: none;
        list-style-position: outside;
        margin-right: 4px;
        margin-top: 4px;
        padding-right: 4px;
        border-right: 2px solid #959595;
    }
}

@media only screen and (min-width: 600px){
    div.card {
        flex-direction: row;
        max-width: 700px;
        min-height: 100px;
    }

    div.grid-item-img{
        flex-basis: 25%;
    }

    div.article-items {
        display: inline-block;
        padding: 2em 0 .25em 0;
        font-size: 0.8rem;
        font-family: 'Roboto', sans-serif;
        line-height: 1;
    }

    div.grid-item-text {
        flex-basis: 75%;
    }

    div.grid-item-text h3.article-header {
        display: block;
        margin: 0;
        font-family: 'Roboto', sans-serif;
        font-weight: 500;
        color: rgba(0,0,0,.87);
        line-height: 1.285em;
        font-size: 1.2rem;
        width: 90%;
    }

    div.article-items>.article-item {
        display: inline-block;
        table-layout: fixed;
        list-style-type: none;
        list-style-position: outside;
        margin-right: 8px;
        margin-top: 4px;
        padding-right: 8px;
        border-right: 2px solid #959595;
    }
}


@media only screen and (min-width: 768px){
    div.card {
        flex-direction: row;
        max-width: 700px;
        min-height: 100px;
    }

    div.grid-item-img{
        flex-basis: 25%;
    }

    div.article-items {
        display: inline-block;
        padding: 2em 0 .25em 0;
        font-size: 1rem;
        font-family: 'Roboto', sans-serif;
        line-height: 1;
    }

    div.grid-item-text {
        flex-basis: 75%;
    }

    div.grid-item-text h3.article-header {
        display: block;
        margin: 0;
        font-family: 'Roboto', sans-serif;
        font-weight: 500;
        color: rgba(0,0,0,.87);
        line-height: 1.285em;
        font-size: 1.3rem;
        width: 90%;
    }

    div.article-items>.article-item {
        display: inline-block;
        table-layout: fixed;
        list-style-type: none;
        list-style-position: outside;
        margin-right: 8px;
        margin-top: 4px;
        padding-right: 8px;
        border-right: 2px solid #959595;
    }
}
