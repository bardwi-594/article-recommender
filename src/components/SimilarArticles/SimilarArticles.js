import React from "react";
import { Link } from 'react-router-dom';
import formatPercent from '../../utils/utils'; /** Used to format the decimal value to percent **/
import moment from 'moment'; /** Used momentjs library to format the date format **/
import "./SimilarArticles.css";

/***Component for the rendering the similarArticles card***/
const SimilarArticles = (props) => {
    const { article } = props; /** Articles Passed down as props**/
    return (
        <div className="list-item">
            {/** Each card links to the source website url from article data **/}
            <Link target="_blank" rel="noreferrer" to={{ pathname: `${article.url}` }}>
                <div className="grid-container card">
                    <div className="grid-item-img">
                        <img alt="complex" src={article.thumbnail} /> {/** Thumbnail image**/}
                    </div>
                    <div className="grid-item-text">
                        <h3 className="article-header">
                            {article.title}   {/** Article Title **/}
                        </h3>
                        <div className="article-items">
                            <div className="article-item similarity">
                                {/** Similarity percent **/}
                                {formatPercent(article.similarity)}% <span>Similar</span>
                            </div>
                            <div className="article-item date-font">
                                {/** Published date **/}
                                {moment(article.published).format('DD.MMM.YY')}
                            </div>
                            <div className="article-item source-font">
                                {/** Source name **/}
                                {article.source_name}
                            </div>
                        </div>

                    </div>
                </div>
            </Link>
        </div>
    )
}

export default SimilarArticles;