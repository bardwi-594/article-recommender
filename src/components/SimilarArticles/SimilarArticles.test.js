import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, render, mount, configure } from 'enzyme';
import SimilarArticles from "./SimilarArticles";
import moment from 'moment';
import formatPercent from "../../utils/utils";

/**Test case for Similar Articles Component**/

configure({ adapter: new Adapter() });
describe('Test for Similar Articles', () => {
    it('renders SimilarArticles component', () => {
        const article = {
            url: "https://www.axios.com",
            similarity: "0.84",
            title: "title",
            published: "2021-02-02d",
            source_name: "Axios",
            thumbnail: "https://images.getnewsbot.com/20775162.jpg"

        }
        expect(shallow(<SimilarArticles article={article} />)).toMatchSnapshot();
    });

});

;
const published = () => {
    const publishedtime = moment();
    console.log(moment())
    return publishedtime;
};

describe('Format', () => {
    it('should return the date in a specific format', () => {
        const wrapper = published().format("DD.MMM.YY");
        expect(wrapper).toEqual("03.Feb.21");
    });

    it('formats a number', () => {
        const similarity = formatPercent(0.8414);
        expect(similarity).toEqual(84);
    })
});