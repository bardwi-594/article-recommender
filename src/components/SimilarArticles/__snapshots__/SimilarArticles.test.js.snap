// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimilarArticles /> renders <SimilarArticles /> component 1`] = `ShallowWrapper {}`;
